scripts/metrics/pytest_metrics_xml_to_csv.py

```diff
@@ -0,0 +1,168 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Scrapes metrics from Pytest xml files and creates csv formatted files.
+"""
+import os.path
+import sys
+import csv
+import argparse
+import xml.etree.ElementTree as xmlElementTree
+import datetime
+
+import ly_test_tools.cli.codeowners_hint as codeowners_hint
+from common import logging, exception
+
+
+TESTING_DIR = 'Testing'
+
+# Setup logging.
+logger = logging.get_logger("test_metrics")
+logging.setup_logger(logger)
+
+# Create the csv field header
+PYTEST_FIELDS_HEADER = [
+    'test_name',
+    'status',
+    'duration_seconds',
+    'sig_owner'
+]
+SIG_OWNER_CACHE = {}
+
+
+def _get_default_csv_filename():
+    """
+    Returns a default filename to be saved in the format of YYYY_MM_DDTHH_mm_SS_pytest.csv
+    :return: The filename as a string
+    """
+    # Format default file name based off of date
+    now = datetime.datetime.isoformat(datetime.datetime.now(tz=datetime.timezone.utc), timespec='seconds')
+    return f"{now.replace('+00:00', 'Z').replace('-', '_').replace('.', '_').replace(':', '_')}_pytest.csv"
```

Comment:
With the possibility of pipelines running in parallel and uploading to the same location, these file names won't be guaranteed to be unique. While including seconds helps reduce collision, I a collision handling mechanism still needs to be considered. I assume the S3 uploader doesn't provide this, so perhaps the file names should use a UUID? (possibly instead of a timestamp? If we really need it, the timestamp info could instead be a metric added into the CSV file)

Reply:
Hm, yeah the uploader won't help here. Unfortunately I think we may need to use uuid to avoid collisions. I was trying to prevent that as they are less readable.

Reply:
I could include microseconds in the timestamped filename, but that also reduces readability and does not guarantee collision handling

Reply:
How about making the folder a Uuid?

Reply:
With that suggestion path will look like:
`o3de-metrics\csv\development\Week08\1a9207c0-7a31-4e5b-a20d-775639de8c77\2023_02_23T21_27_35Z_pytest.csv`

I'm not sure if it would be better to just have:
`o3de-metrics\csv\development\Week08\1a9207c0-7a31-4e5b-a20d-775639de8c77.csv`

Reply:
How about separating the Uuids using <dot>, then?
(Also Week08 seems redundant if the timestamp is part of the filename, as that can be calculated from the timestamp)
`o3de-metrics\csv\development\2023_02_23T21_27_35Z.1a9207c0-7a31-4e5b-a20d-775639de8c77.pytest.csv

Reply:
> Perhaps the first 8 digits of the Uuid would be sufficient. `o3de-metrics\csv\development\2023_02_23T21_27_35Z.1a9207c0.pytest.csv`

I proposed separating YYYY/MM/DD into folders above, but I also like this file naming proposal. Salting with 8 UUID digits seems sufficient since we're unlikely to get more than a few collisions on the same second.

There are likely to be hundreds of files generated per day, and saving everything in the same `development` folder would result in needing to regex filter a large set of files. Pre-partitioning by day provides report generators simpler collection of every relevant file in known folders.

Reply:
New format will be:
`o3de-metrics\csv\development\2023\02\23\f56aa22b.2023_02_23T22_00_58Z_pytest.csv`